homeassistant/util/template.py

```diff
@@ -20,7 +25,11 @@ def render_with_possible_json_value(hass, template, value):
     except ValueError:
         pass
 
-    return render(hass, template, variables)
+    try:
+        return render(hass, template, variables)
+    except TemplateError:
+        _LOGGER.exception('Error parsing value')
+        return value
```

Comment:
I guess I see this as a helper method on top of render.

Can you think of a use case where the caller would want to do something
else?

On Fri, Dec 11, 2015, 19:51 Philip Lundrigan <notifications@github.com>
wrote:

> In homeassistant/util/template.py
> <https://github.com/balloob/home-assistant/pull/734#discussion_r47426520>:
>
> > @@ -20,7 +25,11 @@ def render_with_possible_json_value(hass, template, value):
> >      except ValueError:
> >          pass
> >
> > -    return render(hass, template, variables)
> > +    try:
> > +        return render(hass, template, variables)
> > +    except TemplateError:
> > +        _LOGGER.exception('Error parsing value')
> > +        return value
>
> Just curious, why not allow the exception to propagate and let the caller
> decide what to do with the exception?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/balloob/home-assistant/pull/734/files#r47426520>.
>